gqlalchemy/query_builders/memgraph_query_builder.py

```diff
@@ -79,8 +80,97 @@ def load_csv(self, path: str, header: bool, row: str) -> "DeclarativeBase":
 
         return self
 
+    def construct_subgraph_query(
+        self, node_label: Optional[str] = None, relationship_types: Optional[Union[str, Iterable[str]]] = None
+    ) -> str:
+        """Constructs a MATCH query defining a subgraph using zero or one node labels and relationship types.
+
+        Args:
+            node_label: A string representing a node label to be used in this subgraph.
+            relationship_types: A string or iterable of types of relationships used in the subgraph.
+
+        Returns:
+            a string representing a MATCH query for a path with given node labels and relationship types.
+        """
+        node = CypherNode(node_label)
+
+        # directed is OK only if we have single label, homogenous graphs (current implementation)
+        query = f" MATCH p={node}{CypherRelationship(relationship_types, RelationshipDirection.RIGHT)}{node}"
+
+        return query
```

Comment:
what's the decision then if source node label is different from target node, like in the example here:
https://memgraph.com/docs/mage/how-to-guides/run-a-subgraph-module

Reply:
Actually the example is invalid though, since I used tweets for everything

Reply:
Added support for two labels. It is even easier to make it support more than two, so I don't see why not. 
But from the usage side, when going for multiple, you would also have to define relationship types for each relationship connecting the nodes, and would need to keep track of the order you input it, so you would pretty much know how to make those in cypher.